ckanext/xloader/plugin.py

```diff
@@ -233,9 +233,14 @@ def get_helpers(self):
 def _should_remove_unsupported_resource_from_datastore(res_dict):
     if not toolkit.asbool(toolkit.config.get('ckanext.xloader.clean_datastore_tables', False)):
         return False
+    has_url_type = True
+    try:
+        has_url_type = toolkit.asbool(res_dict.get('url_type'))
+    except ValueError:
+        pass
     return (not XLoaderFormats.is_it_an_xloader_format(res_dict.get('format', u''))
             and (res_dict.get('url_type') == 'upload'
-                or res_dict.get('url_type') == '')
+                or not has_url_type)
```

Comment:
This should just use `not res_dict.get('url_type')`. `asbool` is not the right function for the job, see https://github.com/ckan/ckan/issues/7909

Reply:
Ah I see okay, Done!